### PR TITLE
Make notfound-catch locale-independent on Windows

### LINKED ISSUE #35: API check-node-version v4.0.1 terminates with throwing an error on Windows/nodist
Environment
------------

| item     | version |
| ---      | --- |
|OS        |Windows 10|
|node.js   |10.15.2|
|npm       |6.1.0|
|check-node-version|4.0.1|

How to reproduce?
---------------------------

1. Install [nodist](https://github.com/nullivex/nodist) on Windows
1. Install node v10
    ```bash
    nodist + 10    
    ```

1. run check-node-version
    ```bash
    check-node-version --print
    ```

What happens? 
---------------------

- An error is occured about npx
    ```
    check-node-version\cli.js:45
        throw err;
        ^
    
    Error: Command failed: "npx --version"
    
    stderr:
    'npx' �́A�����R�}���h�܂��͊O���R�}���h�A
    ����\�ȃv���O�����܂��̓o�b�` �t�@�C���Ƃ��ĔF������Ă��܂���B
    
    
    original error message:
    Command failed: npx --version
    'npx' �́A�����R�}���h�܂��͊O���R�}���h�A
    ����\�ȃv���O�����܂��̓o�b�` �t�@�C���Ƃ��ĔF������Ă��܂���B
    
    
        at exec (C:\Users\Yuki\Documents\Projects\Growi\growi\node_modules\check-node-version\index.js:58:21)
        at ChildProcess.exithandler (child_process.js:301:5)
        at ChildProcess.emit (events.js:189:13)
        at maybeClose (internal/child_process.js:970:16)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
    ```
- Error message is broken

What is the expected result? 
-------------------------------------------

- Terminate without throwing an error that is the same behavior at v3.3.0 or below
- (Avoid garbled characters if possible)

Note
----

- nodist does not bundle npx
- In the case of check-node-version v3.3.0 
    ```
    check-node-version --print
    node: 10.15.2
    npm: 6.1.0
    'npx' �́A�����R�}���h�܂��͊O���R�}���h�A
    ����\�ȃv���O�����܂��̓o�b�` �t�@�C���Ƃ��ĔF������Ă��܂���B
    yarn: 1.13.0
    ```

I would like to use check-node-version programmable API, v4.0.1 throws an error, therefore the first argument of callback func 'error' is filled and I couldn't get the result.

### PR DESCRIPTION
**Pull request by rasenplanscher**
Fixes #35

@yuki-takei please check this repo out and run `npm test`

@parshap
Please review, merge, and bump.

The diff is somewhat larger than the actual changes because I rearranged `index.js`. In order to make the code reasonably readable, I had to split the main function in two and add another one. Additionally, I pushed the helper functions `normalizeWanted` and `runVersionCommand` down in the file in order to have the export as far up as possible and increase consistency with the new functions.

### CODE REVIEW

**Comment by rasenplanscher** [comment]
@parshap Also note the badges in `README.md`. Please check them when you bump the version, as per d5ae9e70.

File: index.js [hunk 1]
```diff
@@ -23,81 +23,64 @@ const globalPath = originalPath
...
+    const codepage = stdout.match(/\d+/)[0];
+
+    if (codepage === "65001" || codepage === "437") {
+      // need not switch codepage
+      return run(options);
+    }
+
+    // reset codepage before exiting
+    const finalCallback = options.callback;
+    options.callback = (...args) => exec(`chcp ${codepage}`, () => finalCallback(...args));
```
Comment (parshap) [inline index.js]: Should we handle errors here?
```suggestion
    options.callback = (...args) => exec(`chcp ${codepage}`, err => {
      if (err) {
        throw err;
      }
      finalCallback(...args);
    });
```

File: index.js [hunk 2]
```diff
@@ -23,81 +23,64 @@ const globalPath = originalPath
...
+
+    // switch to Unicode
+    exec("chcp 65001", () => run(options));
```
Comment (parshap) [inline index.js]: Same here

```suggestion
    exec("chcp 65001", err => {
      if (err) {
        throw err;
      }
      run(options);
    });
```

File: index.js [hunk 3]
```diff
@@ -23,81 +23,64 @@ const globalPath = originalPath
...
+  // before trying to optimize this function
+  //
+  // `chcp` is used instead of `where` on account of its more extensive availablity
+  // chcp: MS-DOS 6.22+, Windows 95+; where: Windows 7+
+  //
+  // Plus, in order to be absolutely certain, the error message of `where` would still need evaluation.
+
+  exec("chcp", (error, stdout) => {
+    if (error) {
+      throw error;
```
Comment (parshap) [inline index.js]: Is there a reason to `throw` over propagating the error to the callback?
```suggestion
      options.callback(error);
      return;
```

File: index.js (hunk 1 above)
Comment (rasenplanscher) [reply index.js]: Good catch! These errors, unlikely as they are, should definitely be handled. Will do 👍

File: index.js (hunk 3 above)
Comment (rasenplanscher) [reply index.js]: I don't think so. That's what you get when you spot-build code over months 😅 I'll fix it with the others

**Comment by rasenplanscher** [comment]
@parshap ready for re-review 🙂

**Comment by parshap** [comment]
Published as v4.0.2.

**Comment by yuki-takei** [comment]
@rasenplanscher @parshap v4.0.2 works perfectly! Thank you.

**Pull request merged**